### PR TITLE
Fix external id and jbhunt task

### PR DESCRIPTION
**Pull request by mjmiller**
This branch fixes 2 problems with the schema.org json and microdata importers:
     1. Replace rand external id with a unique constant (url). This eliminates the appearance of duplicates in out index.
      2. Adds a test for a minimum set of required properties specified in each schema.org job posting. If posting fails, it is not included in the index.

This branch also adds a new task to import known JSON schema.org feeds. JB Hunt is one such feed. The task should be scheduled to run regularly on heroku using:
      %>  rake jobs:download_and_import_known_json_schema_dot_org_feeds

### CODE REVIEW

File: lib/importers/schema_dot_org_data.rb [hunk 1]
```diff
@@ -71,4 +71,14 @@ def process_organization_name(job_posting)
       return nil
     end
   end    
+
+  def has_minimum_required_props(job_posting)
+    if job_posting.properties
+      required_properties = ['hiringOrganization','datePosted', 'url']
+      required_properties.map {|x| !job_posting.properties[x].blank?}.all?
```
Comment (greggersh) [inline lib/importers/schema_dot_org_data.rb]: nitpick: use "property" as the variable name, not "x".

**Comment by greggersh** [comment]
Looks good; I don't think we're doing rigorous testing of import rake tasks; but if we are, maybe we should have a spec that tests the case where we don't have the minimum required fields to make sure it fails.

File: lib/importers/schema_dot_org_json_data.rb [hunk 2]
```diff
@@ -67,4 +68,9 @@ def process_organization_name(job)
     return nil
   end
 
+  def has_minimum_required_props(job_posting)
+    required_properties = ['hiringOrganization','datePosted', 'url']
+    required_properties.map {|x| !job_posting[x].blank?}.all?
```
Comment (greggersh) [inline lib/importers/schema_dot_org_json_data.rb]: use something more explicit than `x`.

File: lib/tasks/position_openings.rake [hunk 3]
```diff
@@ -68,7 +68,7 @@ namespace :jobs do
       puts "Example YAML file syntax:"
       puts "bloomingtonmn:"
       puts "\ttags: city tag_2"
-      puts "\torganization_id: US-MN:CITY-BLOOMINGTON"
+e     puts "\torganization_id: US-MN:CITY-BLOOMINGTON"
```
Comment (greggersh) [inline lib/tasks/position_openings.rake]: typo?

File: spec/resources/schema_dot_org/test.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<html>    
+<!--<html>
```
Comment (greggersh) [inline spec/resources/schema_dot_org/test.html]: Name this file something more explicit, like 'html_with_schema_dot_org_but_missing_required_elements.html'

**Comment by greggersh** [comment]
see comments above, otherwise looks good. please include your comment ("This branch fixes 2 problems...") at the top in the commit message.

File: lib/tasks/position_openings.rake (hunk 3 above)
Comment (mjmiller) [inline lib/tasks/position_openings.rake]: hmmm...not seeing a typo.  looks like it was just an edit formatting thing that affected that line.  Am i missing something?

Comment (greggersh) [inline lib/tasks/position_openings.rake]: There's an `e` at the start of the line.

File: lib/importers/schema_dot_org_json_data.rb (hunk 2 above)
Comment (mjmiller) [inline lib/importers/schema_dot_org_json_data.rb]: this is odd.  i thought i changed it to "prop."  better check that all my changes made it into this push.

File: lib/tasks/position_openings.rake (hunk 3 above)
Comment (mjmiller) [inline lib/tasks/position_openings.rake]: ack!!!!!   me dummy.

File: spec/resources/schema_dot_org/test.html (hunk 4 above)
Comment (mjmiller) [inline spec/resources/schema_dot_org/test.html]: you know what, the way i did this the same file is used both when required data is present and missing.  (in the latter case i remove required data AFTER reading in the file.)  now that you pointed this out, mine is a stupid way to do it. i'll create separate files for the various cases.

File: lib/importers/schema_dot_org_json_data.rb (hunk 2 above)
Comment (greggersh) [inline lib/importers/schema_dot_org_json_data.rb]: You did, (see file above), this is another occurence. I missed this on my first pass through.

Might be good to refactor into a base class if they are the same function.

File: spec/resources/schema_dot_org/test.html (hunk 4 above)
Comment (greggersh) [inline spec/resources/schema_dot_org/test.html]: Cool, yeah, bits are cheap.

File: lib/importers/schema_dot_org_json_data.rb (hunk 2 above)
Comment (mjmiller) [inline lib/importers/schema_dot_org_json_data.rb]: Phew.  Thought i was going nutso. 

They are similar but different, so i may have to keep them separate.  However, the required properties should be factored out so they reside in one place.

**Comment by mjmiller** [comment]
@greggersh Please review these latest fixes.  If okay I will modify the commit message as requested above while rebasing.

**Comment by greggersh** [comment]
:+1:

**Pull request closed without merging**